### DISCUSSION TITLE
No module named 'selenium.webdriver.common.options'

### DISCUSSION THREAD

**Discussion by Kamran-ov** [seleniumbase/SeleniumBase#2155] [Q&A] [answered]
Hi, I am having a weird issue.

I have made a py script for a client in Windows OS. it works perfectly fine on my PC.
However, when the client is running it in MAC, he gets the below error.
The script does not pass even the importing line:
`from seleniumbase import Driver` 

```
......py", line 5, in <module>
    from seleniumbase import Driver
 ...../s­­eleniumbase/__init__.py", line 13, in <module>
    from seleniumbase.core.browser_launcher import get_driver # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
....../s­­eleniumbase/core/browser_launcher.py", line 14, in <module>
    from selenium.webdriver.common.options import ArgOptions
ModuleNotFoundError: No module named 'selenium.webdriver.common.options'
```

I have below configuration:
OS: Windows 11
Python: 3.9.12 (Anaconda, conda 4.12.)
seleniumbase         4.19.1
selenium: 4.12.0

the client has the below configuration:
OS: MAC 
Python: 3.11 (just python, not anaconda)
seleniumbase         4.19.1
selenium             4.13.0 (tried also with 4.12.0)

Anybody can suggest what can I do to fix this issue?

Thank you

**Comment by mdmintz** [comment, marked as answer]
There's probably multiple environments on your system and you're not using the correct one. You can verify the import works by typing the following in a `python` prompt:

```python
from selenium.webdriver.common.options import ArgOptions
```

It may be that you need to call your script with `python3` instead of `python`, or something like that to make sure you're using the correct environment. That import is used in several places in `selenium`:

https://github.com/search?q=repo%3ASeleniumHQ%2Fselenium%20ArgOptions&type=code

Maybe use both `pip install seleniumbase` and `pip3 install seleniumbase` to make sure that you have `seleniumbase` and `selenium` installed.

**Reply by Kamran-ov** [reply]
Hi, Thank you for the reply,

I have asked the client to send me the pip and pip3 list. 
both contain the below libraries:
seleniumbase 4.19.1
selenium 4.13.0

I have made a test.py  with the below module imports 
```
from selenium import webdriver
print('Imported Selenium Webdriver')
from selenium.webdriver import Chrome
print('Imported Selenium Webdriver Chrome')
from selenium.webdriver import common
print('Imported Webdriver common')
from selenium.webdriver.common import options
print('Imported Webdriver Chrome Options')
from selenium.webdriver.common.options import ArgOptions
print('Imported Webdriver Chrome ArgOptions ')
```

```
Imported Selenium Webdriver
Imported Selenium Webdriver Chrome
Imported Webdriver common
Traceback (most recent call last):
File "...../test.py", line 7, in <module>
    from selenium.webdriver.common import options
ImportError: cannot import name 'options' from 'selenium.webdriver.common'
```

So, I also went to the webdriver.common folder and it does contain an options.py file (below screenshot) 

I already tried to uninstall and re-install Selenium (also 4.12.0)
Any suggestion on what else I can try? 

![image](https://github.com/seleniumbase/SeleniumBase/assets/50695352/8393be9f-da0c-40e6-9625-be611591febe)

**Reply by mdmintz** [reply]
I'm trying to determine exactly what's missing from your selenium installation. Does this line work?

```python
from selenium.webdriver.common.service import utils
```

Also, you can try confirming your `selenium` version at runtime:

```python
from selenium import __version__
print(__version__)
```
